File: 2026POTD/FEBRUARY/28-02-2026/1680.java
class Solution {
    public int concatenatedBinary(int n) {
        int mod=1_000_000_007;
        long ans=0;
        
        for(int i=1;i<=n;i++){
            int len=Integer.toBinaryString(i).length();
            ans=((ans<<len)+i)%mod;
        }
        return (int)ans;
        
    }
}

File: 2026POTD/FEBRUARY/28-02-2026/1680.cpp
class Solution {
public:
    int concatenatedBinary(int n) {
        int bits = 0;
        long long ans = 0;
        int mod = 1e9 + 7;
        for (int i = 1; i <= n; i++) {
            if ((i & (i - 1)) == 0) {
                bits++;
            }
            ans = ((ans << bits) % (mod) + i) % (mod);
        }

        return (int)ans;
    }
};